Source/JavaScriptCore/b3/B3LowerToAir.cpp

```diff
@@ -70,6 +70,57 @@ namespace JSC { namespace B3 {
 
 namespace {
 
+#if CPU(ARM_THUMB2)
+template <typename Narrow, typename Wide>
+struct MaybeWide {
+    MaybeWide() = default;
+    template <typename T, typename Enable = std::enable_if_t<!std::is_same_v<std::remove_reference_t<T>, MaybeWide>>>
+    /* implicit */ MaybeWide(T&& v)
+        : inner(std::forward<T>(v)) {}
+
+    operator bool() const
+    {
+        return inner.index() != 0;
+    }
+
+    void dump(PrintStream& out) const
+    {
+        if (!*this) {
+            out.print("(<none>)");
+        } else if (inner.index() == 1) {
+            out.print("(", std::get<1>(inner), ")");
+        } else {
+            auto const& wide = std::get<2>(inner);
+            out.print("(", wide.hi, ",", wide.lo, ")");
+        }
+    }
+
+    bool isNarrow() const {return std::holds_alternative<Narrow>(inner);}
+    bool isWide() const {return std::holds_alternative<Wide>(inner);}
+    Narrow narrow() const {return std::get<1>(inner);}
+    Wide wide() const { return std::get<2>(inner); }
+
+    std::variant<std::monostate, Narrow, Wide> inner;
+};
+
+struct WideTmp {
+    Tmp lo, hi;
+};
+
+using SomeTmp = MaybeWide<Tmp, WideTmp>;
```

Comment:
Maybe we can make Tmp->MachineTmp, and SomeTmp->LogicalTmp? Or something like that? Let's get a second opinion

Reply:
Yah,  `SomeTmp` is a bit of a placeholder name. Perhaps `SomeTmp` -> `TmpHolder`? To me `MachineTmp` implies there's something special about the `Tmp`, so renaming it feels like it could mislead the reader. More suggestions welcome.